collection/collection.go

```diff
@@ -340,3 +340,39 @@ func (c Collection[K, V]) Reject(closure func(k K, v V) bool) Collection[K, V] {
 		return !closure(k, v)
 	})
 }
+
+func (c Collection[K, V]) When(execute bool, closure func(collection Collection[K, V]) Collection[K, V]) Collection[K, V] {
+	if !execute {
+		return c
+	}
+
+	return closure(c)
+}
+
+func (c Collection[K, V]) WhenEmpty(closure func(collection Collection[K, V]) Collection[K, V]) Collection[K, V] {
+	if !c.IsEmpty() {
+		return c
+	}
+
+	return closure(c)
```

Comment:
Isn't this the same as `return c.When(c.IsEmpty(), c)`?

Reply:
Done